NYPLAudiobookToolkit/UI/AudiobookPlayerViewController.swift

```diff
@@ -315,12 +163,34 @@ let SkipTimeInterval: Double = 15
         })
         return timeLeftAfterChapter
     }
+  
+    @objc func bookmarkWasPressed() {
+      guard let chapterLocation = currentChapterLocation,
+            let businessLogic = audiobookManager.bookmarkBusinessLogic else {
+        return
+      }
+      
+      bookmarkButtonStateLock.lock()
+      defer {
+        bookmarkButtonStateLock.unlock()
+      }
+      
+      if bookmarkButtonOn,
+         let bookmark = businessLogic.bookmarkExisting(at: chapterLocation) {
+        updateBookmarkButton(withState: !bookmarkButtonOn)
+        businessLogic.deleteAudiobookBookmark(bookmark)
+      } else {
+        updateBookmarkButton(withState: !bookmarkButtonOn)
+        businessLogic.addAudiobookBookmark(chapterLocation)
+      }
+    }
 
     @objc public func tocWasPressed(_ sender: Any) {
-        let tocVC = AudiobookTableOfContentsTableViewController(
-            tableOfContents: self.audiobookManager.tableOfContents,
-            delegate: self)
-        self.navigationController?.pushViewController(tocVC, animated: true)
+      let readerPositionVC = AudiobookReaderPositionsVC(
+        bookmarksBusinessLogic: audiobookManager.bookmarkBusinessLogic,
+        tocProvider: audiobookManager.tableOfContents)
```

Comment:
I don't think the `tocProvider` should be part of the `bookmarkBusinessLogic`, it does complicate things since `bookmarkBusinessLogic` already has quite an amount of code so adding code not related to bookmarking is not a great idea. We can have a standalone business logic class for the TOC but it is just a renaming of the existing class and protocol.

Reply:
sounds good, no worries then.